src/ddscxx/include/org/eclipse/cyclonedds/domain/qos/DomainParticipantQosDelegate.hpp

```diff
@@ -92,6 +92,7 @@ template<>
 inline dds::core::policy::EntityFactory&
 DomainParticipantQosDelegate::policy<dds::core::policy::EntityFactory> ()
 {
+    present_ |= QP_ADLINK_ENTITY_FACTORY;
```

Comment:
This I don't really understand: this means that reading the value marks it as being set, which in turn means it gets passed on to the C API, right? That may or may not be wise (I can see arguments either way), but having two getters and only one of them setting the flag seems like a recipe for confusion. Given that the `const` variant can't set the flag, I think that means the non-const version also mustn't set the flag.

Reply:
The idea here was that we are passing a reference to the EntityFactory QoS, so the person receiving the reference can modifies members of the EntityFactory QoS and hereby implicitly set the presence flag for the EntityFactory QoS.